tutorials/preprocessing/plot_15_handling_bad_channels.py

```diff
@@ -0,0 +1,254 @@
+#
+#
+# When to look for bad channels
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# You can start looking for bad channels during the experiment session when the
+# data is being acquired. If you notice any flat or excessively noisy channels,
+# you can note them in your experiment log or protocol sheet. If your system
+# computes on-line averages, these can be a good way to spot bad channels as
+# well. After the data has been collected, you can do a more thorough check for
+# bad channels by browsing the raw data using :meth:`mne.io.Raw.plot`, with any
+# projectors or ICA applied. Finally, you can compute off-line averages (again
+# with projectors, ICA, and EEG referencing disabled) to look for channels with
+# unusual properties. Here's an example of ERP/F plots where the bad channels
+# were not properly marked:
+
+raw2 = raw.copy()
+raw2.info['bads'] = []
+events = mne.find_events(raw2, stim_channel='STI 014')
+epochs = mne.Epochs(raw2, events=events)['2'].average().plot()
+
+###############################################################################
+# The bad EEG channel is not so obvious, but the bad gradiometer is easy to
+# see.
+#
+# Remember, marking bad channels should be done as early as possible in the
+# analysis pipeline. When bad channels are marked in a :class:`~mne.io.Raw`
+# object, the markings will be automatically transferred through the chain of
+# derived object types: including :class:`~mne.Epochs` and :class:`~mne.Evoked`
+# objects, but also :class:`noise covariance <mne.Covariance>` objects,
+# :class:`forward solution computations <mne.Forward>`, :class:`inverse
+# operators <mne.minimum_norm.InverseOperator>`, etc. If you don't notice the
+# badness until later stages of your analysis pipeline, you'll probably need to
+# go back and re-run the pipeline, so it's a good investment of time to
+# carefully explore the data for bad channels early on.
+#
+#
+# Why mark bad channels at all?
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Many analysis computations can be strongly affected by the presence of bad
+# channels. For example, a malfunctioning channel with completely flat signal
+# will have zero channel variance, which will cause noise estimates to be
+# unrealistically low. This low noise estimate will lead to a strong channel
+# weight in the estimate of cortical current, and because the channel is flat,
+# the magnitude of cortical current estimates will shrink dramatically.
+#
+# Conversely, very noisy channels can also cause problems. For example, they
+# can lead to too many epochs being discarded based on signal amplitude
+# rejection thresholds, which in turn can lead to less robust estimation of the
+# noise covariance across sensors. Noisy channels can also interfere with
+# :term:`SSP <projector>` computations, because the projectors will be
+# spatially biased in the direction of the noisy channel, which can cause
+# adjacent good channels to be suppressed. ICA is corrupted by noisy channels
```

Comment:
hmm ... why would this be? I know it's what I heard from @dengemann but ICA doesn't maximize variance but statistical independence. So why should it be affected?

Reply:
I don't fully understand it myself; it seems to mostly be an issue when the noisy channel is a couple orders of magnitude above the other signals.